### DISCUSSION TITLE
Trigger Off-Canvas widget by any link with id or class name - similar to popup windows

### DISCUSSION THREAD

**Discussion by memdigital** [elementor/elementor#27746] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

The new off-canvas widget can only be triggered using an elementor widget with dynamic off-canvas link applied. Which is perfectly fine, however, some more advanced users may wish to trigger off-canvas on toggl using a custom link, or HTML element, like an animated burger menu, in my instance.

### Describe the solution you'd like.

I would like to be able choose a trigger id or class name, same as I use for popup windows, so I can apply that to my custom element.

### Describe alternatives you've considered.

To my knowledge there is no alternatives. If we could have hover states for icons where the icon 'image' changes, that would be useful. That way I can create a burger menu for example with some custom icons, that change when hovering or active. Then I can use your existing dynamic link opener method.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by devops-graphite-bg** [comment]
+1, please add the option "Open By Selector" so we can trigger Off-Canvas elements by any button or link with custom "a" attribute - the current way of dealing with the opening of the selector results in long link displayed in browser once the respective button is clicked - something that can be avoided with "Open By Selector"

**Comment by somesunnymind** [comment]
+1, please add it. At the moment one cannot link a button with an offscreen widget if only one of the two is embedded in a template.

**Comment by rloes** [comment]
+1 This is very needed. Until then I workaround using a shortcode that manually creates elementor_action_hashes:

```php
add_shortcode( 'elementor_action_hash', function ( $atts ) {

		if ( ! class_exists( '\Elementor\Plugin' ) || ! isset( \Elementor\Plugin::$instance->frontend ) ) {
			return '';
		}

		$atts = shortcode_atts( [
			'action'   => '',
			'settings' => '',
		], $atts, 'elementor_action_hash' );

		$action = trim( (string) $atts['action'] );
		if ( $action === '' ) {
			return '';
		}

		// Accept JSON or query-string for settings.
		$settings_raw = trim( (string) $atts['settings'] );
		$settings = [];

		if ( $settings_raw !== '' ) {
			$decoded = json_decode( $settings_raw, true );
			if ( is_array( $decoded ) ) {
				$settings = $decoded;
			} else {
				parse_str( $settings_raw, $parsed );
				if ( is_array( $parsed ) ) {
					$settings = $parsed;
				}
			}
		}

		$hash = \Elementor\Plugin::$instance->frontend->create_action_hash( $action, $settings );

		// Return only the hash text.
		return esc_html( $hash );
	} );
```

Then you could make a link open a offcanvas by:

``` [elementor_action_hash action="off_canvas:toggle" settings="id=486c4a4&displayMode=toggle"] ```
or
``` [elementor_action_hash action="off_canvas:toggle" settings='{"id":"486c4a4", "displayMode":"toggle"}'] ```
